### PR TITLE
Add accessors to internals

### PR DESCRIPTION
**Pull request by markusbattarbee**
Used for creating batch kernels which can with one launch operate on umpteen different hash maps. Pass the kernel a pointer to the hashmap itself, and it can then act upon the contents in a limited fashion (such as doing a bulk reset)

### CODE REVIEW

File: include/hashinator/hashinator.h [hunk 1]
```diff
@@ -463,6 +463,28 @@ class Hashmap {
...
+      return _mapInfo;
+   }
+   template <bool warn = true>
+   HASHINATOR_HOSTDEVICE hash_pair<KEY_TYPE, VAL_TYPE>* expose_bucketdata() noexcept {
+      if constexpr(warn) {
+         printf("Warning, exposing Hashmap internal bucket data!\n");
+      }
+      return buckets.data();
+   }
+   HASHINATOR_HOSTDEVICE inline KEY_TYPE expose_emptybucket() const noexcept {
```
Comment (kstppd) [inline include/hashinator/hashinator.h]: There is already a method for this right?
https://github.com/kstppd/hashinator/blob/426a1c8924332a1025246503565d6206182a8743/include/hashinator/hashinator.h#L449

File: include/hashinator/hashinator.h [hunk 2]
```diff
@@ -463,6 +463,28 @@ class Hashmap {
...
+      return EMPTYBUCKET;
+   }
+   HASHINATOR_HOSTDEVICE inline KEY_TYPE expose_tombstone() const noexcept {
```
Comment (kstppd) [inline include/hashinator/hashinator.h]: Same here

File: include/hashinator/hashinator.h [hunk 3]
```diff
@@ -463,6 +463,28 @@ class Hashmap {
       }
    }
 
+   // Dangerous methods for exposing internals
```
Comment (kstppd) [inline include/hashinator/hashinator.h:468]: Not a fan of these but this has been discussed.

**Review by kstppd — CHANGES_REQUESTED** [review]
I added some comments but I think you can remove the empty bucket and tombstone methods because those already existed.

**Comment by markusbattarbee** [comment]
OK, I updated this by:
a) removing those two exposes
b) improving threadsafety
c) streamlining device insertions and making them templated for skipping existing values
d) added split wrappers for atomic Min and Max so we get rid of some ugly casting.
Passes tests on my machine.

**Review by kstppd — APPROVED** [review]
Passes on my machine as well

**Pull request merged**